src/charm.py

```diff
@@ -22,25 +22,23 @@
 from charms.prometheus_k8s.v0.prometheus_remote_write import PrometheusRemoteWriteProvider
 from charms.prometheus_k8s.v0.prometheus_scrape import MetricsEndpointProvider
 from ops.charm import CharmBase
-from ops.framework import StoredState
 from ops.main import main
 from ops.model import ActiveStatus, BlockedStatus, WaitingStatus
 from ops.pebble import Error as PebbleError
 from ops.pebble import Layer, PathError, ProtocolError
-from parse import search  # type: ignore
 
 MIMIR_CONFIG = "/etc/mimir/mimir-config.yaml"
 MIMIR_DIR = "/mimir"
-RULES_DIR = f"{MIMIR_DIR}/rules"
 
-logger = logging.getLogger(__name__)
+# Ruler dirs cannot overlap, otherwise mimir complains:
+# error validating config: the configured ruler data directory "/mimir/rules" cannot overlap with
+# the configured ruler storage local directory "/mimir/rules"; please set different paths, also
+# ensuring one is not a subdirectory of the other one
+RULES_DIR = f"{MIMIR_DIR}/rules"
+RULER_DATA_DIR = f"{MIMIR_DIR}/ruler_data"
```

Comment:
@rbarry82 this is a leftover from trial and error. Any reason to keep it?

Reply:
Well, we _will_ want it when we stop using local

Reply:
Ok. I think I'll remove it for now then. At the current pace the config schema may change by that point anyway :)